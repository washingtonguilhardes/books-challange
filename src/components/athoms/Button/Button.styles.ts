import { makeStyles, Theme } from '@material-ui/core/styles';

import { ButtonProps } from './Button.types';

export const useButtonStyles = makeStyles<Theme, ButtonProps>(() => {
  return {
    root({ textColor }) {
      return {
        color: textColor ?? '',
      };
    },
  };
});
