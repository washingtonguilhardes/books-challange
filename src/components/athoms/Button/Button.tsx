import React from 'react';

import { default as MUIButton } from '@material-ui/core/Button';

import clsx from 'clsx';

import { useButtonStyles } from './Button.styles';
import { ButtonProps } from './Button.types';

export function Button(props: ButtonProps) {
  const { textColor, ...others } = props;
  const classes = useButtonStyles({ textColor, ...props });
  return <MUIButton {...others} className={clsx(classes.root, others.className)} />;
}
