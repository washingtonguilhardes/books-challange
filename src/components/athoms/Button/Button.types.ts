import { ButtonProps as MUIButtonProps } from '@material-ui/core/Button';

export interface ButtonProps extends MUIButtonProps {
  textColor?: string;
}
