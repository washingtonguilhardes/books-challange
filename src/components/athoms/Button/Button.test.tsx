import React from 'react';

import { render } from '@testing-library/react';

import { Button } from './Button';

describe('Button', () => {
  it('should match with snapshot', () => {
    const { container } = render(<Button>teste</Button>);
    expect(container).toMatchSnapshot();
  });
});
