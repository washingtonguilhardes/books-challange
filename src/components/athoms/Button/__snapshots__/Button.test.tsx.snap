// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match with snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-root-1 makeStyles-root-2"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      teste
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
